controllers/openstackmachine_controller.go

```diff
@@ -334,6 +347,138 @@ func GetPortIDs(ports []infrav1.PortStatus) []string {
 	return portIDs
 }
 
+// reconcileFloatingAddressFromPool reconciles the floating IP address from the pool.
+// It returns the IPAddressClaim and a boolean indicating if the IPAddressClaim is ready.
+func (r *OpenStackMachineReconciler) reconcileFloatingAddressFromPool(ctx context.Context, scope *scope.WithLogger, openStackMachine *infrav1.OpenStackMachine, openStackCluster *infrav1.OpenStackCluster) (*ipamv1.IPAddressClaim, bool, error) {
+	if openStackMachine.Spec.FloatingAddressFromPoolRef == nil {
+		return nil, false, nil
+	}
+	var claim *ipamv1.IPAddressClaim
+	claim, err := r.getOrCreateIPAddressClaimForFloatingAddress(ctx, scope, openStackMachine, openStackCluster)
+	if err != nil {
+		conditions.MarkFalse(openStackMachine, infrav1.FloatingAddressFromPoolReadyCondition, infrav1.FloatingAddressFromPoolErrorReason, clusterv1.ConditionSeverityInfo, "Failed to reconcile floating IP claims: %v", err)
+		return nil, true, err
```

Comment:
Do we want to return `true` as the second result even on error case? Then the condition in line 523 can be simpler.

Reply:
I think the best practice is to return zero values in case of errors so *new(bool) / false, but explicitly returning false on a bool stating if we need to wait, I don't remember the reason why it's a "shouldWait" instead of a "isReady" I suspect it was initially passed to Request.Requeue but was removed because we don't need to Requeue because we are watching the related objects anyway.

With the current returns 523 would only need to check the bool, but not having an nilcheck on the error feels like a trap for the future and I think the lint want us to do the check the error(?)

I'm down for flipping the bool to specify ready instead if we think that would be clearer